### DISCUSSION TITLE
useRequest 更多功能

### DISCUSSION THREAD

**Discussion by bowencool** [alibaba/hooks#1699] [Ideas]
比如 [debounceAsyncResult](https://bowencool.github.io/async-utilities/functions/debounceAsyncResult/readme.html)、[throttleAsyncResult](https://bowencool.github.io/async-utilities/functions/throttleAsyncResult/readme.html)

**Comment by bowencool** [comment]
[@bowencool/async-utilities](https://github.com/bowencool/async-utilities) 这个库也是我写的，已经在我司内部 vue3 Composition API 得到集成，单测覆盖率 100%，如果有什么顾虑或需求，也可以直说

**Comment by brickspert** [comment]
> 

不好意思哈，我没太看明白，你说的 debounce/throttle，与目前 ahooks 已有的 debounce/throttle 有什么区别呢？

**Reply by bowencool** [reply]
<img width="1035" alt="image" src="https://user-images.githubusercontent.com/20217146/175026495-4683eb07-9ee9-480d-b02f-c8ea2d295e8b.png">
<img width="1020" alt="image" src="https://user-images.githubusercontent.com/20217146/175026607-b9143456-0499-413b-aeff-61cf9cb34ff0.png">

**Comment by bowencool** [comment]
给个 vue 版的示例：https://bowencool.github.io/vhooks/hooks/use-async-state/readme.html
